{"cells": [{"cell_type": "markdown", "metadata": {"editable": false}, "source": ["# Genome Assembly\n", "\n", "The algorithmic reconstruction of contiguous chromosomes from short fragments of DNA.\n", "\n", "<img style=\"margin-left: 0px\" src=\"http://rosalind.info/static/img/topics/pictures/genome-assembly.jpg?v=1526042457\">", "\n", "\n", "Rosalind link: [Genome Assembly](http://rosalind.info/problems/topics/genome-assembly/)\n", "\n", "### Import modules"]}, {"cell_type": "code", "execution_count": null, "metadata": {"editable": false}, "outputs": [], "source": ["import os\n", "import sys\n", "from itertools import permutations\n", "import numpy as np\n", "import pandas as pd\n", "from Bio.Seq import Seq\n", "from Bio import SeqIO\n", "from Bio.Alphabet import generic_rna\n", "\n", "print('DONE!')"]}, {"cell_type": "markdown", "metadata": {}, "source": ["### Genome Assembly as Shortest Superstring\n", "\n", "Rosalind link: [Genome Assembly as Shortest Superstring](http://rosalind.info/problems/long/)\n"]}, {"cell_type": "code", "execution_count": null, "metadata": {}, "outputs": [], "source": ["\n", "# TODO", "\n"]}, {"cell_type": "markdown", "metadata": {}, "source": ["### Error Correction in Reads\n", "\n", "Rosalind link: [Error Correction in Reads](http://rosalind.info/problems/corr/)\n"]}, {"cell_type": "code", "execution_count": null, "metadata": {}, "outputs": [], "source": ["\n", "# TODO", "\n"]}, {"cell_type": "markdown", "metadata": {}, "source": ["### Constructing a De Bruijn Graph\n", "\n", "Rosalind link: [Constructing a De Bruijn Graph](http://rosalind.info/problems/dbru/)\n"]}, {"cell_type": "code", "execution_count": null, "metadata": {}, "outputs": [], "source": ["\n", "# TODO", "\n"]}, {"cell_type": "markdown", "metadata": {}, "source": ["### Genome Assembly with Perfect Coverage\n", "\n", "Rosalind link: [Genome Assembly with Perfect Coverage](http://rosalind.info/problems/pcov/)\n"]}, {"cell_type": "code", "execution_count": null, "metadata": {}, "outputs": [], "source": ["\n", "# TODO", "\n"]}, {"cell_type": "markdown", "metadata": {}, "source": ["### Genome Assembly Using Reads\n", "\n", "Rosalind link: [Genome Assembly Using Reads](http://rosalind.info/problems/gasm/)\n"]}, {"cell_type": "code", "execution_count": null, "metadata": {}, "outputs": [], "source": ["\n", "# TODO", "\n"]}, {"cell_type": "markdown", "metadata": {}, "source": ["### Assessing Assembly Quality with N50 and N75\n", "\n", "Rosalind link: [Assessing Assembly Quality with N50 and N75](http://rosalind.info/problems/asmq/)\n"]}, {"cell_type": "code", "execution_count": null, "metadata": {}, "outputs": [], "source": ["\n", "# TODO", "\n"]}, {"cell_type": "markdown", "metadata": {}, "source": ["### Genome Assembly with Perfect Coverage and Repeats\n", "\n", "Rosalind link: [Genome Assembly with Perfect Coverage and Repeats](http://rosalind.info/problems/grep/)\n"]}, {"cell_type": "code", "execution_count": null, "metadata": {}, "outputs": [], "source": ["\n", "# TODO", "\n"]}], "metadata": {"kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "mimetype": "text/x-python", "name": "python", "nbconvert_exporter": "python", "pygments_lexer": "ipython3", "version": "3.7.2"}}, "nbformat": 4, "nbformat_minor": 2}